packages/web-forms/src/components/controls/RankControl.vue

```diff
@@ -0,0 +1,206 @@
+<script setup lang="ts">
+import { inject, ref, watch } from 'vue';
+import { VueDraggable } from 'vue-draggable-plus';
+import type { RankItem, RankNode } from '@getodk/xforms-engine';
+import ControlText from '@/components/ControlText.vue';
+import ValidationMessage from '@/components/ValidationMessage.vue';
+
+interface RankControlProps {
+	readonly question: RankNode;
+}
+
+interface RankDraggableOption {
+	value: string;
+	label: string;
+}
+
+const props = defineProps<RankControlProps>();
+const HOLD_DELAY = 60; // Delay in ms to hold an item before dragging, avoids accidental reordering on swipe.
+const options = ref<RankDraggableOption[]>([]);
+const touched = ref(false);
+const submitPressed = inject<boolean>('submitPressed');
+const highlight = {
+	index: ref(null),
+	timeoutID: null,
+};
+
+const transformOptions = (rankOptions: RankItem[]) => {
+	if (options.value.length) {
+		options.value.forEach((option) => {
+			option.label = props.question.getValueLabel(option.value);
+		});
+		return;
+	}
+
+	options.value = rankOptions.map((option) => {
+		return { value: option.value, label: props.question.getValueLabel(option.value) };
+	});
+};
+
+watch(props.question.currentState.valueOptions, transformOptions, { immediate: true });
+
+const setValues = () => {
+	touched.value = true;
+	props.question.setValues(options.value.map((option) => option.value));
+};
+
+const setHighlight = (index: number) => {
+	highlight.index.value = index;
+
+	if (highlight.timeoutID) {
+		clearTimeout(highlight.timeoutID);
+		highlight.timeoutID = null;
+	}
+
+	if (highlight.index.value !== null) {
+		highlight.timeoutID = setTimeout(() => setHighlight(null), 1000);
+	}
+};
+
+const moveUp = (index: number) => {
+	const newPosition = index - 1;
+	if (newPosition < 0) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const moveDown = (index: number) => {
+	const newPosition = index + 1;
+	if (newPosition >= options.value.length) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const swapItems = (index: number, newPosition: number) => {
+	setHighlight(index);
+	const temp = options.value[index];
+	options.value[index] = options.value[newPosition];
+	options.value[newPosition] = temp;
+	setHighlight(newPosition);
+};
+</script>
+
+<template>
+	<ControlText :question="question" />
+
+	<VueDraggable
+		v-model="options"
+		:id="question.nodeId"
+		:delay="HOLD_DELAY"
+		:disabled="question.currentState.readonly"
+		ghost-class="fade-moving"
+		class="rank-control"
+		@update="setValues">
+		<div
+			v-for="(option, index) in options"
+			:key="option.value"
+			class="rank-option"
+			:class="{ 'moving': highlight.index.value === index }"
+			tabindex="0"
+			@keydown.up.prevent="moveUp(index)"
+			@keydown.down.prevent="moveDown(index)">
+			<div class="rank-label">
+				<svg xmlns="http://www.w3.org/2000/svg" width="25" height="25" viewBox="0 0 768 768">
```

Comment:
These SVGs are temporary, I have [another PR](https://github.com/getodk/web-forms/pull/279) that will make this nicer

Reply:
Gonna resolve this now since that's a place to discuss in more detail!